### PR TITLE
Add validation to illness end date

### PR DESCRIPTION
**Pull request by mreis-ch**
[BI-8946](https://companieshouse.atlassian.net/browse/BI-8946)

Create new validator class IllnessEndDateValidator. This extends DateValidator and includes functionality to check whether the end date is before the start date.
Update IllnessEndDateController to use new validator class, IllnessEndDateValidator.
Update start date in future error message to date in future error message to make now generic

### Work checklist

- [x] Tests added where applicable
- [ ] UI changes look good on mobile
- [ ] UI changes meet accessibility criteria

### Merge instructions

We are committed to keeping commit history clean, consistent and linear. To achieve this commit should be structured as follows:

```
<type>[optional scope]: <description>
```

and contain the following structural elements:

- fix: a commit that patches a bug in your codebase (this correlates with PATCH in semantic versioning),
- feat: a commit that introduces a new feature to the codebase (this correlates with MINOR in semantic versioning),
- BREAKING CHANGE: a commit that has a footer `BREAKING CHANGE:` introduces a breaking API change (correlating with MAJOR in semantic versioning). A BREAKING CHANGE can be part of commits of any type,
- types other than `fix:` and `feat:` are allowed, for example `build:`, `chore:`, `ci:`, `docs:`, `style:`, `refactor:`, `perf:`, `test:`, and others,
- footers other than `BREAKING CHANGE: <description>` may be provided.

### CODE REVIEW

File: src/controllers/IllnessEndDateController.ts [hunk 1]
```diff
@@ -1,9 +1,9 @@
 import { SessionMiddleware } from 'ch-node-session-handler';
 import { controller } from 'inversify-express-utils';
 import moment from 'moment';
+import { IllnessEndDateValidator } from './validators/IllnessEndDateValidator';
```
Comment (mouhajer-ch) [inline src/controllers/IllnessEndDateController.ts:4]: For consistency better use the src application path 
`import { IllnessEndDateValidator } from 'app/controllers/validators/IllnessEndDateValidator';`

File: src/controllers/validators/IllnessEndDateValidator.ts [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+        const validationResult: ValidationResult = await super.validate(request);
+
+        const session: Session | undefined = request.session;
+
+        if (!session) {
+            throw SESSION_NOT_FOUND_ERROR;
+        }
+
+        const appData: ApplicationData = session.getExtraData<ApplicationData>(APPLICATION_DATA_KEY)
+            || { appeal: {} as Appeal, navigation: { permissions: [] } } as ApplicationData;
```
Comment (mouhajer-ch) [inline src/controllers/validators/IllnessEndDateValidator.ts:41]: if appData is undefined I will throw an Error instead, something like

    if(!appData) {
        throw APPLICATION_DATA_UNDEFINED;
    }

File: src/controllers/IllnessEndDateController.ts (hunk 1 above)
Comment (mreis-ch) [reply src/controllers/IllnessEndDateController.ts:4]: Ah yes, ok 👍

File: src/controllers/validators/IllnessEndDateValidator.ts (hunk 2 above)
Comment (mreis-ch) [reply src/controllers/validators/IllnessEndDateValidator.ts:41]: Will do

Comment (mreis-ch) [reply src/controllers/validators/IllnessEndDateValidator.ts:41]: Just want to check. Line 40, doesn't this line always set appData? I'm guessing if the getExtraData returns nothing then an empty appData object is created?

Comment (mouhajer-ch) [reply src/controllers/validators/IllnessEndDateValidator.ts:41]: At this page `appData` should not be empty, better fail quick

**Review by mouhajer-ch — APPROVED** [review]

**Pull request merged**